pilot/pkg/serviceregistry/kube/controller/ambient/multicluster.go

```diff
@@ -336,11 +298,114 @@ func (a *index) buildGlobalCollections(
 		opts,
 	)
 
-	WorkloadAddressIndex := krt.NewIndex[networkAddress, model.WorkloadInfo](GlobalWorkloads, "networkAddress", networkAddressFromWorkload)
-	WorkloadServiceIndex := krt.NewIndex[string, model.WorkloadInfo](GlobalWorkloads, "service", func(o model.WorkloadInfo) []string {
+	workloadNetworkServiceIndex := krt.NewIndex[string, model.WorkloadInfo](GlobalWorkloads, "network;service", func(o model.WorkloadInfo) []string {
+		res := make([]string, 0, len(o.Workload.Services))
+		for svc := range o.Workload.Services {
+			res = append(res, strings.Join([]string{o.Workload.Network, svc}, ";"))
```

Comment:
Similar to above, does this get easier if we just add Network to WorkloadInfo?

Reply:
Instead of this weird string representation, could we just Fetch the workload network?